### PR TITLE
32679 fixed FindPeaks crash when the number of bins are not sufficient

### LINKED ISSUE #32679: FindPeaks crashes if run on histograms containing a single bin
This is clearly not a realistic histogram to try finding peaks in - the reason I noticed this is that I had a workspace containing many single bin histograms and I had forgotten to transpose it before running FindPeaks. It shouldn't crash Mantid though...

### PR DESCRIPTION
**Pull request by warunawickramasingha**
### Description of work
Fixed a crash on `FindPeaks` algorithm when run on histograms containing a single bin by raising an error message on this condition.

Fixes #32679. 

### To test:
1. Run the below code and the algorithm should not crash instead should throw a meaningful error
```
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

ws = CreateWorkspace(OutputWorkspace='test', DataX='1', DataY='0,0,0,2,5,2,0,0,0,0', NSpec=10)

peaks_ws = FindPeaks(ws)
```

2. Errors should be raised even when the `FindPeaks` algorithm is run from Algorithms window on a similar workspace without crashing the workbench.

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Review by sf1919 — DISMISSED** [review]
Please can you check whether the following warnings apply to this PR? The others I think are being fixed elsewhere

GNU C Compiler (Linux)
- [FindPeaks.cpp](https://builds.mantidproject.org/job/pull_requests-conda-linux/7236/gcc/fileName.141045931/)

Clang Warnings (OSX)
- [FindPeaks.cpp](https://builds.mantidproject.org/job/pull_requests-conda-osx/5761/clang/fileName.141045931/)

File: Framework/Algorithms/src/FindPeaks.cpp [hunk 1]
```diff
@@ -403,6 +403,13 @@ void FindPeaks::findPeaksUsingMariscotti() {
   if (!(w % 2))
     ++w;
 
+  if (!m_dataWS->isRaggedWorkspace() && (m_dataWS->blocksize() <= static_cast<size_t>(w))) {
+    std::stringstream errss;
+    errss << "Block size of the workspace should be greater than:" << w;
+    g_log.warning(errss.str());
```
Comment (RichardWaiteSTFC) [inline Framework/Algorithms/src/FindPeaks.cpp:409]: It's not necessary to print a warning as well as throw an error with the same message - it prints this
```
Block size of the workspace should be greater than:601
Error in execution of algorithm FindPeaks:
Block size of the workspace should be greater than:601
```

**Review by RichardWaiteSTFC — CHANGES_REQUESTED** [review]
Nice fix and thanks for adding a test. It would be good to have this check done in the validation but there is no input validation run before the exec in this algorithm and it doesn't seem worth it just for this fix. Just a minor thing then good to approve!

File: Framework/Algorithms/src/FindPeaks.cpp (hunk 1 above)
Comment (warunawickramasingha) [reply Framework/Algorithms/src/FindPeaks.cpp]: Thanks for noting that! Did the changes as requested.

**Review by RichardWaiteSTFC — DISMISSED** [review]
Thanks for the change looks good!

**Review by RichardWaiteSTFC — APPROVED** [review]

**Pull request merged**